mlflow/entities/gateway_secrets.py

```diff
@@ -4,7 +4,7 @@
 from mlflow.entities._mlflow_object import _MlflowObject
 
 
-@dataclass
+@dataclass(frozen=True)
```

Comment:
This means last_updated_at can't be updated as well, if it's never modified then it's fine